daemon/api_systems.go

```diff
@@ -180,9 +190,48 @@ type systemActionRequest struct {
 
 	client.SystemAction
 	client.InstallSystemOptions
+	client.CreateSystemOptions
 }
 
 func postSystemsAction(c *Command, r *http.Request, user *auth.UserState) Response {
+	contentType := r.Header.Get("Content-Type")
+	mediaType, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		mediaType = "application/json"
+	}
+
+	switch mediaType {
+	case "application/json":
+		return postSystemsActionJSON(c, r)
+	case "multipart/form-data":
+		return postSystemsActionForm(c, r, params)
+	default:
+		return BadRequest("unexpected media type %q", mediaType)
+	}
+}
+
+func postSystemsActionForm(c *Command, r *http.Request, contentTypeParams map[string]string) Response {
+	boundary := contentTypeParams["boundary"]
+	mpReader := multipart.NewReader(r.Body, boundary)
+	form, errRsp := readForm(mpReader)
+	if errRsp != nil {
+		return errRsp
+	}
+
+	action := form.Values["action"]
```

Comment:
there should be a deferred call to `form.RemoveAllExcept(nil)` when postSystemsActionForm returns with error beyond this point.